src/main/java/org/assertj/core/api/Assumptions.java

```diff
@@ -0,0 +1,757 @@
+   *
+   * @param actual the actual value.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assumeThat(Throwable actual) {
+    return enhanceAsAssumption(ThrowableAssert.class, Throwable.class, actual);
+  }
+
+  /**
+   * Allows to capture and then assume on a {@link Throwable} (done easier when used with Java 8 lambdas).
+   *
+   * @param shouldRaiseThrowable The {@link ThrowingCallable} or lambda with the code that should raise the throwable.
+   * @return the created assumption for assertion object.
+   */
+  @CheckReturnValue
+  public static AbstractThrowableAssert<?, ? extends Throwable> assumeThatThrownBy(ThrowingCallable shouldRaiseThrowable) {
+    return enhanceAsAssumption(ThrowableAssert.class, Throwable.class, catchThrowable(shouldRaiseThrowable));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <ASSERTION, ACTUAL> ASSERTION enhanceAsAssumption(Class<ASSERTION> assertionType,
+                                                                   Class<ACTUAL> actualType,
+                                                                   Object actual) {
+    Enhancer enhancer = new Enhancer();
+    enhancer.setSuperclass(assertionType);
+    enhancer.setCallback(new MethodInterceptor() {
+      @Override
+      public Object intercept(Object o, Method method, Object[] args, MethodProxy methodProxy) throws Throwable {
+        try {
+          Object result = methodProxy.invokeSuper(o, args);
+          if (result != o && result instanceof AbstractAssert) {
+            return enhanceAsAssumption(result);
+          }
+          return result;
+        } catch (AssertionError e) {
+          throw createAssumptionViolatedException(e);
+        }
+      }
+    });
+
+    return (ASSERTION) enhancer.create(new Class[] { actualType }, new Object[] { actual });
+  }
+
+  private static RuntimeException createAssumptionViolatedException(AssertionError e) throws ReflectiveOperationException {
+    try {
+      return createAssumptionViolatedException(Class.forName("org.junit.AssumptionViolatedException"), e);
+    } catch (ClassNotFoundException junitClassNotFound) {
+      try {
+        return createAssumptionViolatedException(Class.forName("org.testng.SkipException"), e);
+      } catch (ClassNotFoundException testngClassNotFound) {
+        throw new IllegalStateException("Assumptions require JUnit or TestNG in the classpath");
+      }
+    }
+  }
+
+  private static RuntimeException createAssumptionViolatedException(Class<?> exceptionClass,
+                                                                    AssertionError e) throws ReflectiveOperationException {
+    return (RuntimeException) exceptionClass.getConstructor(String.class, Throwable.class)
+                                            .newInstance("assumption failed due to assertion error", e);
+  }
```

Comment:
keep the assertion error message to explain why the assumption was not met, I'm not sure we should mention assertion here as it is not about asserting but assuming, maybe something like `String.format("assumption was not met, error was:%n%s", e.getMessage())`